core/crypto/sm3.c

```diff
@@ -50,6 +50,16 @@ void sm3_init(struct sm3_context *ctx)
 	ctx->state[7] = 0xB0FB0E4E;
 }
 
+static inline uint32_t rotl(uint32_t val, int shift)
+{
+	if (shift == 0)
+		return val;
+
+#define SHL(x, n)	((x) << (n))
```

Comment:
This define in the middle of a function looks kinda weird. How about moving the `rotl()` function definition just above the `#define ROTL(x, n) rotl((x), (n))` line?

Reply:
I tried to match the current style with macros defined in the middle of the function.
I need `SHL()` in `rotl()`
```
core/crypto/sm3.c: In function ‘rotl’:
core/crypto/sm3.c:58:16: error: implicit declaration of function ‘SHL’ [-Werror=implicit-function-declaration]
   58 |         return SHL(val, shift & 0x1F) | (val >> (32 - (shift & 0x1F)));
      |                ^~~
core/crypto/sm3.c:58:16: warning: nested extern declaration of ‘SHL’ [-Wnested-externs]
```

Reply:
In the middle of the `sm3_process()` function?